pkg/api/api.go

```diff
@@ -240,6 +245,109 @@ func (a *API) RegisterDistributor(d *distributor.Distributor, pushConfig distrib
 	distributorpb.RegisterDistributorServer(a.server.GRPC, d)
 
 	a.RegisterRoute("/api/v1/push", push.Handler(pushConfig.MaxRecvMsgSize, a.sourceIPs, a.cfg.SkipLabelNameValidationHeader, a.cfg.wrapDistributorPush(d)), true, false, "POST")
+	// Endpoint to handle requests for starting of block backfilling.
+	// Starting the backfilling of a block means to verify that the backfill can go ahead.
+	// In practice this means to check that the block isn't already in block storage.
+	a.RegisterRoute("/api/v1/backfill/{tenant:[0-9]+}/{block}", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Do not specify the tenant in the path. The tenant always comes from the `X-Scope-OrgID` request header. Get it from the context using `tenant.TenantID(ctx)`.

Reply:
@pracucci thanks, will do.

Reply:
Done.